### PR TITLE
[PLA-11268] Override StatusDetail.__repr__.

### PR DESCRIPTION
**Pull request by anoto-moniz**
# Citrine Python PR

To make status_detail behave more like the deprecated status_info, the StatusDetail object should override __repr__, so the messages (and levels) appear when printed and listed.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

File: src/citrine/resources/status_detail.py [hunk 1]
```diff
@@ -30,3 +30,6 @@ def __init__(self, *, msg: str, level: Union[str, StatusLevelEnum]):
 
     def __str__(self):
         return f"[{self.level.upper()}] {self.msg}"  # pragma: no cover
+
+    def __repr__(self):
+        return f"<{str(self)}>"  # pragma: no cover
```
Comment (kroenlein) [inline src/citrine/resources/status_detail.py:35]: Why no cover?

**Review by kroenlein — APPROVED** [review]
It's fine, but you really should have test coverage.

File: src/citrine/resources/status_detail.py (hunk 1 above)
Comment (anoto-moniz) [reply src/citrine/resources/status_detail.py:35]: Laziness. But you're right, I'll go add a test for both `__str__` and `__repr__`.

**Review by kroenlein — APPROVED** [review]
LGTM

**Pull request merged**